### DISCUSSION TITLE
Cascading scenario requires SDongle but this causes constant drop-outs of comms w/ RS485

### DISCUSSION THREAD

**Discussion by JohnMcLear** [wlcrs/huawei_solar#595] [Connecting to the inverter]
I want to use this integration but if you have multiple inverters you will have a bad user experience so I'm open to alternative ideas..

There are several tickets and pieces of documentation to explain this but I'll massively simplify it...

If you have multiple inverters and a Power meter:

- To have stable comms you need to NOT use the SDongle and instead use the direct RS485, this works great.
- To have a working Power meter (that doesn't make the inverters go into power limited mode[basically crippling the devices]) you need to use the SDongle.

From Huawei:
> Are these 3 inverters in cascading scenario?
> 
> If yes, normally the inverters cannot be cascaded without an SDongle.

So what can I do?

**Comment by Roving-Ronin** [comment]
@JohnMcLear 
I'm running 2 x M1 6kW inverters in a cascade with a DTSU-666H and sDongle. 
As required, sDongle inserted into the primary inverter and the DTSU and Secondary inverter then connected via RS485.

WiFi AP is a UniFi nanoHD ~4m away in the roof space of the garage, with this having an SSID for IoT devices that is limited to 2.4Ghz / 802.11Ghz WiFi and rate limiting allowing devices that connect at slow speeds (i.e. 1Mb even), in addition to a seperate SSID for normal users that allow 5Ghz and all the fancy WiFi things. IoT WiFi has its own subnet and VLAN, with the HA server on a different subnet/VLAN.

This installation has been rock solid with no communication / drop out etc issues, that as you suggest are the WLCRS integration fault. 

IMHO I would suggest you should go back to basics and check your RS485 wiring, DTSU connection and sDongle functioning (as well as making sure the inverters are configured to have differing ID numbers for modbus). Get the basics working for FusionSolar Portal and then see about adding in the WLCRS integration (and entering those modbus ID# in the correct order).
At the same time I'd also be looking at your WiFi network and making sure you don't have it setup with options that see slow connections automatically disconnected, 2.4Ghz connections automatically forced to try connect as 5Ghz connections (Band Steering), Fast Roaming, etc.

If you use UniFi for example here's what I run, with all the 'fancy' stuff turned off for IoT devices:

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/c3cc1cce-75e6-4958-8177-c16bd54517b3)

**Reply by JohnMcLear** [reply]
Infra is fine buddy, I have 1Gb connection to a Unifi AP that is about 2M from SDongle location and I've load tested/ping tested/latency tested the SDongle connection and it's fully functioning at a TCP/IP level with the only failures happening at a software level within the Inverter.

> I'm running 2 x M1 6kW inverters in a cascade with a DTSU-666H and sDongle.

Are you reporting back to Fusion Solar / and/or a management system or no?

**Reply by Roving-Ronin** [reply]
@JohnMcLear my inverter cascade concurrently reports to FusionSolar portal and HA/WLCRS, using the sDongle (WiFi with DHCP reserved IP on IoT VLAN/subnet, with HA on different VLAN/Subnet).  DTSU-666H and 38 600W optimisers on the 2 inverters and all show in HA & FusionSolar.

Inverters:  2 x SUN2000-6KTL-M1 running V100R001C00SPC155 
    - Note: SPC158 is the latest, but haven't updated as last few updates are just OEM rebranding and other country settings

sDongle: SDongleA-05  running  V200R022C10SPC108


IMHO I'd work on getting it all working and showing in FusionSolar via the sDongle, then once that is working, write down the inverters ID# 1,2,3 (or whatever random numbers the installer used) and try and add into HA using the WLCRS integration (entering the ID's in the order of Inverter1 / Inverter2 / Inverter3 and not numerically sequential ordering).

**Reply by JohnMcLear** [reply]
I had it working fine w/ Fusion Solar & sDongle but that causes drops into HA.

I already did what you suggested, I feel like maybe you are missing a lot of backstory here.  If you look through the repo/discussions it explains a lot of my journey!  I even wrote the documentation in parts :P

**Comment by JohnMcLear** [comment]
I've been gathering more debug data and should be able to send a pr soon.

My infra is fine, as you probably have seen there have been quite a few commits land to work on Comms stability.
